callbacks/multi_callbacks_handler.go

```diff
@@ -0,0 +1,139 @@
+package callbacks
+
+import (
+	"context"
+	"sync"
+
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/schema"
+)
+
+var _ Handler = &MultiCallbacksHandler{}
+
+type MultiCallbacksHandler struct {
+	Callbacks []Handler
+}
+
+func executeCallbacksParallelly(callbacks []Handler, f func(Handler)) {
+	wg := sync.WaitGroup{}
+	wg.Add(len(callbacks))
+	for _, handler := range callbacks {
+		go func() {
+			defer wg.Done()
+			f(handler)
```

Comment:
Thanks @recht! Agreed. I'm working on a fresh commit which I plan to push out sometime this weekend - for the time being I did away with wanting to parallelize because of the possible need to deterministically merge context (I went with the approach where context is passed back from callback handlers - this changes the handler contract but allows handlers to have a way to manage state during the run of a chain).